### DISCUSSION TITLE
[BUG] Details link is inconsistent for checks reported by GitHub Apps

### DISCUSSION THREAD

**Discussion by webknjaz** [community/community#24387] [Apps, API and Webhooks]
<p>Example: <a href="https://github.com/tox-dev/tox/pull/1272#partial-pull-merging" rel="nofollow noopener">https://github.com/tox-dev/tox/pull/1272#partial-pull-merging</a></p>
<p>When you scroll to the bottom and look at the box with status reports from integrations, there’s a <a href="https://github.com/apps/psf-chronographer" rel="nofollow noopener">Details</a> link. My app reports a check called " <strong>Timeline protection</strong>" via Checks API.</p>
<p>This details link is supposed to take the user to the Checks tab in the PR (<a href="https://github.com/tox-dev/tox/pull/1272/checks?check_run_id=159224582" rel="nofollow noopener">https://github.com/tox-dev/tox/pull/1272/checks?check_run_id=159224582</a>).</p>
<p>Sometimes it works properly, but often it links to the app installation page (<a href="https://github.com/apps/psf-chronographer" rel="nofollow noopener">https://github.com/apps/psf-chronographer</a>) which is weird.</p>

**Comment by lee-dohm** [comment]
<p>I’m unable to reproduce the problem you’re describing using the example link you provided.</p>
<p>Are you still experiencing this issue?</p>

**Comment by bobvanderlinden** [comment]
<p>I’m having the same issue. I have a GitHub App that creates check runs. The check runs have a <code>output</code> field and there I supply some additional information. I do _not_ supply the checkrun with a <code>details_url</code>, as I want the output to be visible in GitHub.</p>
<p>Sometimes all check runs in my check suite will have a details link directing to the home-page of the GitHub App (<a href="https://github.com/bobvanderlinden/check-reporter" rel="nofollow noopener">https://github.com/bobvanderlinden/check-reporter</a>). Other times a link to the actual check run details page is used: <a href="https://github.com/aio-libs/aiohttp/pull/3804/checks?check_run_id=163078825" rel="nofollow noopener">https://github.com/owner/repo/pull/1234/checks?check_run_id=163078825</a>. This last case is what I always intend to happen.</p>
<p>What is also interesting is that the checks and their output/details do always show up correctly in the Checks tab of a PR.</p>

**Comment by webknjaz** [comment]
<p>It is still broken</p>

**Comment by billyvg** [comment]
<p>Do you still have this problem? I’m seeing this issue as well.</p>

**Comment by bobvanderlinden** [comment]
<p>The problem is solved for me. I <em>think</em> the problem was that I created a check suite myself, whereas GitHub also creates a check suite for my app automatically. I can imagine that the incorrect details link was happening because the checks suite was manually created after the check was submitted.</p>
<p>That said, this is all speculation. I can close this report for me, but if this is still a problem for others I can leave it open, as my fix was the result of not understanding the documentation well enough.</p>

**Comment by billyvg** [comment]
<p>Yeah I’m not sure exactly what’s going on, but for now I’ve explicitly set <code>details_url</code> to undefined.</p>

**Comment by webknjaz** [comment]
<p>I do not create check suites, only check runs: <a href="https://github.com/sanitizers/chronographer-github-app/blob/04c024e1c4b7642c1cdcc3d70a9880cb285f0696/chronographer/event_handlers.py#L160-L169" rel="nofollow noopener">https://github.com/sanitizers/chronographer-github-app/blob/04c024e1c4b7642c1cdcc3d70a9880cb285f0696/chronographer/event_handlers.py#L160-L169</a>. And it’s still happening sometimes.</p>

**Comment by billyvg** [comment, marked as answer]
<p>Ahh - be aware if you use the <code>Refined Github</code> browser extension, there is a setting that may cause this:</p>
<p><img src="https://user-images.githubusercontent.com/79684/181084414-5f756e77-c782-4e0d-86b1-2207c4cdef5f.png" alt="image" data-base62-sha1="ncPIU7jHdiEgXt6VApemNx9FDym" width="532" height="68"></p>
<p>Now I’m facing the opposite problem where I want the user to skip the GitHub Checks screen but it’s not (and I am supplying a <code>details_url</code>).</p>

**Comment by webknjaz** [comment]
<aside class="quote no-group" data-username="billyvg" data-post="10" data-topic="14145">
<div class="title">
<div class="quote-controls"></div>
 billyvg:</div>
<blockquote>
<p>Ahh - be aware if you use the <code>Refined Github</code> browser extension, there is a setting that may cause this:</p>
</blockquote>
</aside>
<p>Good point. I do use it. But the question stands: why is it inconsistent? Also, <em>I think</em> I also saw this behavior in an incognito session and some HTTP response in DevTools.<br>
OTOH disabling this feature in Refined GitHub fixes it for me.</p>
<p>I guess another real problem is that my app links its installation page via details URL and that’s where Refined GitHub grabs it from…</p>
